src/gtk/spinctrl.cpp

```diff
@@ -67,16 +52,57 @@ gtk_value_changed(GtkSpinButton* spinbutton, wxSpinCtrlGTKBase* win)
 
 extern "C" {
 static void
-gtk_changed(GtkSpinButton* spinbutton, wxSpinCtrl* win)
+gtk_changed(GtkSpinButton*, wxSpinCtrl* win)
 {
     wxCommandEvent event( wxEVT_TEXT, win->GetId() );
     event.SetEventObject( win );
-    event.SetString(gtk_entry_get_text(GTK_ENTRY(spinbutton)));
+    event.SetString(win->GetTextValue());
     event.SetInt(win->GetValue());
     win->HandleWindowEvent( event );
 }
 }
 
+//-----------------------------------------------------------------------------
+// "input" and "output"
+//-----------------------------------------------------------------------------
+
+extern "C"
+{
+
+static gint
+wx_gtk_spin_input(GtkSpinButton*, gdouble* val, wxSpinCtrlGTKBase* win)
+{
+    switch ( win->GTKInput(val) )
+    {
+        case wxSpinCtrl::GTKInput_Error:
+            return GTK_INPUT_ERROR;
+
+        case wxSpinCtrl::GTKInput_Default:
+            return FALSE;
+
+        case wxSpinCtrl::GTKInput_Converted:
+            return TRUE;
+    }
+
+    wxFAIL_MSG("unreachable");
+    return FALSE;
+}
+
+static gint
```

Comment:
Oops, indeed, thanks. I've fixed the return type of the "input" handler to correctly use `gint` rather than `gboolean` but didn't realize that "output" was symmetrically wrong in the other direction...